### PR TITLE
Add bounded network request correlation

### PR DESCRIPTION
**Pull request by C76GN**
## Summary

为可选 Network 扩展加入有界请求关联器，统一处理回复匹配、超时和断线清理。项目负责协议编码、回复识别与业务结果解释；Tracker 按真实传输 peer 和完整请求 ID 完成一次性句柄。

## Contract and Risk

- Change type: feat
- Consumer-visible behavior: 新增 `GFNetworkRequestTracker`、`GFNetworkRequestHandle`、`GFNetworkRequestResult`，支持同步回复、单调超时、本地取消和会话隔离；结果保存并返回隔离副本。回复校验及副本准备均计入等待时间，准备完成后再次检查截止时间才提交成功。
- API or extension contract: 主线程显式驱动会话、`tick()` 与 `dispose()`。默认容量 64，上限 4096；超时为 1–86,400,000 ms；响应限制深度 16、节点数 4096、估算字节数 65,536。`gf.network` 的 `extension_version` 为 `7.1.0`，新增 API 使用 `@since unreleased`。
- Persisted data, package, protocol, or ProjectSettings impact: 没有新增持久化格式、ProjectSettings 或固定协议字段。项目须完整回显不透明请求 ID，并使用传输回调的真实 peer。
- Failure, cancellation, rollback, concurrency, and ownership impact: 登记后同步发送，终态只提交一次；批量关闭先结算再通知，监听者关闭可中止会话重开。Tracker 不拥有传输，调用方负责生命周期接线与定期推进；超时不能抢占同步发送，取消不撤销远端操作，匹配成功不代表业务成功。
- Compatibility and migration: 兼容新增，无需迁移现有网络调用。共享校验器新增字节预算默认关闭。GF 根开发身份保持 `12.0.0-dev.0`。

## Verification

- [x] Focused tests cover the changed behavior and failure path — 4 scripts, 205 tests, 1866 assertions passed; zero orphans and unhandled warnings. 包括 33 项请求关联测试；新增副本准备后到期的回归在旧实现失败、修复后通过。
- [x] `python tools/gf_maintenance.py check --suite quick --json-output build/network-request-quick-complete.json` — 初始功能提交通过 30 项检查；本次修复由下述 Full 覆盖同一检查集合。
- [x] GDScript warning and LSP gates are clean — standalone and final Full strict LSP scanned 1410 files with zero diagnostics and timeouts; Full reload warning gate passed.
- [x] Generated API or knowledge output is fresh when its source changed — owning generators rerun; private fix introduces no generated diff.
- [x] `python tools/gf_maintenance.py check --suite full --json-output build/network-request-review-complete.json` — 40 checks passed, including 375 GUT scripts / 7299 tests / 89502 assertions, all integration checks, and source snapshot integrity.
- [x] Draft PR feedback completed through `GF draft gate` for the initial feature commit.
- [x] Ready PR CI completed through `GF repository policy` and `GF merge gate` for the latest commit — all four framework shards, Windows process supervision, and the validation marker for the frozen base passed in run `34457229831`.

GitHub 托管附加审查 [`github-advanced-security`](https://github.com/C76GN/gf-framework/actions/runs/34457232207/job/102806360186) 在创建审查请求时报告 `400 The requested model is not supported`，GitHub 不允许重试该托管 workflow。同一提交的 CodeQL、Analyze (actions/python) 已通过。

聚焦测试由维护工具 `run_command` 监督，使用仓库 GUT CLI 和 pre/post lifecycle hooks。以下展开实际运行的参数（Bash；完成项目导入，`godot` 指向 Godot 控制台可执行文件）：

```sh
mkdir -p ai_analysis/godot_logs
godot --headless --path . \
  --log-file ai_analysis/godot_logs/network_request_review_focused.log \
  -s res://tests/gf_core/support/gf_gut_cli.gd \
  -gconfig= \
  -gtest=res://tests/gf_core/extensions/network/test_gf_network_request_tracker.gd \
  -gtest=res://tests/gf_core/extensions/network/test_gf_network_extension.gd \
  -gtest=res://tests/gf_core/extensions/network/test_gf_network_sync_coordinator.gd \
  -gtest=res://tests/gf_core/maintenance/test_gdscript_parse_validation.gd \
  -gpre_run_script=res://tests/gf_core/support/gf_gut_pre_run_hook.gd \
  -gpost_run_script=res://tests/gf_core/support/gf_gut_post_run_hook.gd \
  -gexit
```

## Documentation and Release

- [x] Relevant guide and `[未发布]` changelog entries are updated.
- [x] Development version impact is correct; the review fix does not alter the intended SemVer line.
- [x] This PR does not create a tag or publish a release.

指南包含同步完成的等待方式、Session 信号接线与释放示例、截止时间及响应预算；Catalog、Reference、知识索引和测试归属已同步。

### CODE REVIEW

File: addons/gf/extensions/network/runtime/gf_network_request_tracker.gd [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+		return false
+	if _expire(handle):
+		return false
+	var report: Dictionary = _VALUE_VALIDATOR_SCRIPT.validate(response, _RESPONSE_LIMITS)
+	if _expire(handle):
+		return false
+	if not GFVariantData.get_option_bool(report, "ok"):
+		var _finished: bool = _finish(handle, GFNetworkRequestResult.STATUS_INVALID_RESPONSE)
+		return false
+	return _finish(handle, GFNetworkRequestResult.STATUS_RECEIVED, response)
```
[the remark this answers is not in the document]
Comment (C76GN) [reply addons/gf/extensions/network/runtime/gf_network_request_tracker.gd]: 已修复：先完成 `GFNetworkRequestResult` 的隔离副本准备，再检查截止时间，最后由不发通知的 `_commit_result()` 提交；最后一次检查之后不会再次复制响应。若复制期间到期，原等待项由超时路径完成，`receive_reply()` 返回 `false`，且不会保留或发布 `received` 结果。批量关闭仍先全部提交再通知。

新增两个可控时钟回归，分别覆盖准备快照后恰好到期，以及截止前成功并保持副本隔离。到期用例在旧实现中失败，修复后通过，并检查终态只通知一次、晚到回复/取消不改变结果。

验证：4 脚本 / 205 项聚焦测试通过；最终 Full 的 40 项检查全部通过（7299 项测试 / 89502 个断言），严格 LSP 扫描 1410 个文件且零诊断。
